packages/prettier-plugin-java/src/printers/classes.js

```diff
@@ -511,7 +506,8 @@ class ClassesPrettierVisitor {
 
   throws(ctx) {
     const exceptionTypeList = this.visit(ctx.exceptionTypeList);
-    return join(" ", [ctx.Throws[0], exceptionTypeList]);
+    const throwsDeclaration = join(" ", [ctx.Throws[0], exceptionTypeList]);
+    return group(indent(rejectAndConcat([softline, throwsDeclaration])));
```

Comment:
We could add an extra indent here, so that the throws + exceptionTypeList is not at the same level of the rest of the body, WDYT ?

Reply:
Hmm, would that be the only place in prettier-java where a double-indent is used? There is a similar issue with a really long class declaration, for example:

```java
public class MyReallyLongClassName
  extends SomeOtherReallyLongClass
  implements SomeReallyLongInterface {
  private Field field;

  public Field someMethodWithAReallyLongName()
    throws IOException {
    return field;
  }
}
```

I wonder if the fix in both cases could be to force a newline if the class signature or throws clause linebreaks? This would look like:

```java
public class MyReallyLongClassName
  extends SomeOtherReallyLongClass
  implements SomeReallyLongInterface {

  private Field field;

  public Field someMethodWithAReallyLongName()
    throws IOException {

    return field;
  }
}
```

Reply:
There are other places where we could add some double indents indeed.

It might indeed be better to add an extra blank line, but it is more difficult to implement, IMO, as the formatting of the class body is not in the same node as the inheritance/interface declaration.

Reply:
Would it be easier if class bodies always started with an empty line? (basically #271) 

I guess that wouldn't help the method case though, since you wouldn't want methods to unconditionally start with an empty line

Reply:
Also getting back to the original suggestion, double indent (or continuation indent) is pretty common in Java formatters but doesn't seem to be idiomatic in prettier as far as I can tell (this issue didn't seem to go anywhere https://github.com/prettier/prettier/issues/5865). 

If we introduced the idea of continuation indent, I think we'd want to apply it consistently. Off the top of my this would include class signatures:
```java
public class MyReallyLongClassName
  ->extends SomeOtherReallyLongClass
  ->implements SomeReallyLongInterface {
```

Method signatures:
```java
public void myMethod(
  ->String argOne
  ->String argTwo
  ->String argThree
) {}
```

Method call chains:
```java
return someCollectionType
  ->.stream()
  ->.filter(x -> x != null)
  ->.collect(Collectors.toList());
```

And probably a bunch of other statements as well. This is definitely one way to go, but would be a pretty massive formatting change at this point. Might be better to move discussion to an issue if that's under consideration